src/common/checksum.c

```diff
@@ -139,6 +140,36 @@ void daos_csummer_destroy(struct daos_csummer **obj)
 	*obj = NULL;
 }
 
+int
+daos_csummer_copy_inplace(struct daos_csummer *dst, struct daos_csummer *src)
+{
+	int rc = 0;
+
+	dst->dcs_algo = src->dcs_algo;
+	dst->dcs_chunk_size = src->dcs_chunk_size;
+	dst->dcs_srv_verify = src->dcs_srv_verify;
+	dst->dcs_skip_key_calc = src->dcs_skip_key_calc;
+	dst->dcs_skip_key_verify = src->dcs_skip_key_verify;
+	dst->dcs_skip_data_verify = src->dcs_skip_data_verify;
+
+	if (dst->dcs_algo->cf_init)
+		rc = dst->dcs_algo->cf_init(&dst->dcs_ctx);
```

Comment:
Not sure if the cf_init()/cf_destroy() is expensive, but it sounds to me they are not supposed to be called on each fetch request?

Reply:
This is what initializes the internal resources for the csummer. I'm not quite sure how to get around this.

Reply:
I guess a question I have is that if csummer is per-container, can we do something to separate the buffer used from the other stuff such that we don't have to initialize callbacks and things all over the place where the only thing that really changes is the buffer and maybe some internal variables